File: frontend/src/types/canvas.ts
export type CanvasObjectType =
  | 'circle'
  | 'rect'
  | 'ellipse'
  | 'line'
  | 'arrow'
  | 'triangle'
  | 'diamond'
  | 'star'
  | 'sticky'
  | 'process'
  | 'image_placeholder'
  | 'text'

export interface CanvasObjectState {
  object_key: string
  object_type: CanvasObjectType
  name?: string
  properties: Record<string, unknown>
}

export interface CanvasState {
  width: number
  height: number
  objects: CanvasObjectState[]
}

File: backend/internal/models/canvas.go
package models

import "time"

type CanvasState struct {
	ProjectID uint64         `json:"project_id"`
	Width     int            `json:"width"`
	Height    int            `json:"height"`
	Objects   []CanvasObject `json:"objects"`
	UpdatedAt *time.Time     `json:"updated_at,omitempty"`
}

type CanvasObject struct {
	ObjectKey  string         `json:"object_key"`
	ObjectType string         `json:"object_type"`
	Name       string         `json:"name,omitempty"`
	Properties map[string]any `json:"properties"`
}
